### DISCUSSION TITLE
Is there rook client with golang?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#10095] [Q&A]
Is there rook client with golang?

**Comment by zhucan** [comment]
@leseb

**Comment by subhamkrai** [comment]
@zhucan we don't have a rook go client but we have a go-ceph client https://github.com/ceph/go-ceph

**Comment by zhucan** [comment]
we need rook go client, not need go-ceph client @subhamkrai

**Reply by subhamkrai** [reply]
oh  I thought of something else, maybe this helps https://pkg.go.dev/github.com/rook/rook/pkg/client/clientset/versioned?utm_source=gopls

**Comment by zhucan** [comment]
Yes, But we need it can be import by other repo.

**Reply by subhamkrai** [reply]
yes!

**Comment by zhucan** [comment]
@travisn Is there a plan?

**Reply by travisn** [reply]
The golang API for creating the rook CRs is already available where @subhamkrai mentioned. Golang APIs can be imported easily across repos so there is nothing else needed for us to make available.

Internally we use the rook client in the tests, you can search the rook code for:
```go
rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
```

And to see how it is called, search for `RookClientset`.

**Comment by zhucan** [comment]
@travisn Thanks.

**Comment by zhucan** [comment]
@travisn 
![image](https://user-images.githubusercontent.com/13997135/164170764-b4094700-97af-4bd3-843a-63dec35a6c74.png)
Use the  "rookclient "github.com/rook/rook/pkg/client/clientset/versioned"" by our repo ,  exec go mod tidy failed.

**Reply by subhamkrai** [reply]
is this rook master?

**Reply by zhucan** [reply]
v1.8.5

**Reply by subhamkrai** [reply]
can you share `go env` output?

**Reply by zhucan** [reply]
module xxxx.ai/smd
replace (
github.com/googleapis/gnostic => github.com/googleapis/gnostic v0.4.1
github.com/kubernetes-incubator/external-storage => github.com/libopenstorage/external-storage v0.20.4-openstorage-rc3
github.com/portworx/sched-ops => github.com/portworx/sched-ops v0.20.4-openstorage-rc3
)

exclude (
github.com/kubernetes-incubator/external-storage v0.20.4-openstorage-rc2
k8s.io/client-go v1.4.0
k8s.io/client-go v1.5.0
k8s.io/client-go v1.5.1
k8s.io/client-go v1.5.2
k8s.io/client-go v10.0.0+incompatible
k8s.io/client-go v11.0.0+incompatible
k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
k8s.io/client-go v12.0.0+incompatible
k8s.io/client-go v2.0.0+incompatible
k8s.io/client-go v2.0.0-alpha.1+incompatible
k8s.io/client-go v3.0.0+incompatible
k8s.io/client-go v3.0.0-beta.0+incompatible
k8s.io/client-go v4.0.0+incompatible
k8s.io/client-go v4.0.0-beta.0+incompatible
k8s.io/client-go v5.0.0+incompatible
k8s.io/client-go v5.0.1+incompatible
k8s.io/client-go v6.0.0+incompatible
k8s.io/client-go v7.0.0+incompatible
k8s.io/client-go v8.0.0+incompatible
k8s.io/client-go v9.0.0+incompatible
k8s.io/client-go v9.0.0-invalid+incompatible
)

Add above to the go.mod, it ok for me. And our go version is v1.16, but the "client-go" version is 0.23.5, the k8s(v1.23) need go's version is v1.17, bu  rook only support v1.16.

**Comment by zhucan** [comment]
https://github.com/libopenstorage/secrets/issues/50 The same errors.
